src/tribler/core/utilities/sqlite_utils.py

```diff
@@ -0,0 +1,241 @@
+# classes that replaces `sqlite3.Cursor` and `sqlite3.Connection` classes respectively. If the `connect` function or
+# any Connectoin or Cursor method is called and the database is corrupted, the database is marked as corrupted and
+# the DatabaseIsCorrupted exception is raised. It should be handled by terminating the Tribler Core with the exit code
+# EXITCODE_DATABASE_IS_CORRUPTED (99). After the Core restarts, the `handle_db_if_corrupted` function checks the
+# presense of the database corruption marker and handles it by removing the database file and the corruption marker.
+# After that, the database is recreated upon the next attempt to connect to it.
+
+
+logger = logging.getLogger(__name__)
+
+
+class DatabaseIsCorrupted(Exception):
+    pass
+
+
+@contextmanager
+def _handling_malformed_db_error(db_filepath: Path):
+    # Used in all methods of Connection and Cursor classes where the database corruption error can occur
+    try:
+        yield
+    except Exception as e:
+        if _is_malformed_db_exception(e):
+            _mark_db_as_corrupted(db_filepath)
+            raise DatabaseIsCorrupted(str(db_filepath)) from e
+        raise
+
+
+def _is_malformed_db_exception(exception):
+    return isinstance(exception, sqlite3.DatabaseError) and 'malformed' in str(exception)
+
+
+def _mark_db_as_corrupted(db_filepath: Path):
+    # Creates a new `*.is_corrupted` marker file alongside the database file
+    marker_path = get_corrupted_db_marker_path(db_filepath)
+    marker_path.touch()
+
+
+def get_corrupted_db_marker_path(db_filepath: Path) -> Path:
+    return Path(str(db_filepath) + '.is_corrupted')
+
+
+def handle_db_if_corrupted(db_filename: Union[str, Path]):
+    # Checks if the database is marked as corrupted and handles it by removing the database file and the marker file
+    db_path = Path(db_filename)
+    marker_path = get_corrupted_db_marker_path(db_path)
+    if marker_path.exists():
+        _handle_corrupted_db(db_path)
+
+
+def _handle_corrupted_db(db_path: Path):
+    # Removes the database file and the marker file
+    if db_path.exists():
+        logger.warning(f'Database file was marked as corrupted, removing it: {db_path}')
+        db_path.unlink()
+
+    marker_path = get_corrupted_db_marker_path(db_path)
+    if marker_path.exists():
+        logger.warning(f'Removing the corrupted database marker: {marker_path}')
+        marker_path.unlink()
+
+
+def connect(db_filename: str, **kwargs) -> sqlite3.Connection:
+    # Replaces the sqlite3.connect function
+    kwargs['factory'] = Connection
+    with _handling_malformed_db_error(Path(db_filename)):
+        return sqlite3.connect(db_filename, **kwargs)
+
+
+class CursorBase(sqlite3.Cursor):
+    # This class simplifies testing of the Cursor class by allowing mocking of base class methods.
+    # Direct mocking of sqlite3.Cursor methods is not possible because they are C functions.
+    pass
+
+
+class Cursor(CursorBase):
```

Comment:
Could be spit to separate files.

Reply:
I replaced `sqlite_utils` with `db_corruption_handling.base` and `db_corruption_handling.sqlite_replacement`. Now, the module `db_corruption_handling.sqlite_replacement` is sufficiently short to not split it further.